lib/Core/PTree.cpp

```diff
@@ -36,6 +36,8 @@ PTree::PTree(ExecutionState *initialState)
   initialState->ptreeNode = root.getPointer();
 }
 
+PTree::~PTree() { delete root.getPointer(); }
```

Comment:
Why is this required? As soon as the last state is removed the root node should have been removed as well (PTree::remove).

Reply:
PTree::remove isn't called during the test I mentioned.

Reply:
> isn't called during the test I mentioned.

Indeed, hence it requires `not-asan`.

However, your `cleanUpStates()` removes all states and that's why I'm asking.